Mathlib/LinearAlgebra/Matrix/BilinearForm.lean

```diff
@@ -193,6 +193,23 @@ theorem BilinForm.toMatrix_apply (B : BilinForm R₁ M₁) (i j : n) :
     BilinForm.toMatrix b B i j = B (b i) (b j) :=
   LinearMap.toMatrix₂_apply _ _ B _ _
 
+theorem BilinForm.dotProduct_toMatrix_mulVec (B : BilinForm R₁ M₁) (x y : n → R₁) :
+    x ⬝ᵥ (BilinForm.toMatrix b B) *ᵥ y = B (b.equivFun.symm x) (b.equivFun.symm y) := by
+  simp only [dotProduct, mulVec_eq_sum, op_smul_eq_smul, Finset.sum_apply, Pi.smul_apply,
+    transpose_apply, toMatrix_apply, smul_eq_mul, mul_sum, Basis.equivFun_symm_apply, map_sum,
+    map_smul, coeFn_sum, LinearMap.smul_apply]
+  rw [Finset.sum_comm]
+  refine Finset.sum_congr rfl (fun i _ ↦ Finset.sum_congr rfl fun j _ ↦ ?_)
+  ring
+
+lemma BilinForm.apply_eq_dotProduct_toMatrix_mulVec (B : BilinForm R₁ M₁) (x y : M₁) :
+    B x y = (b.repr x) ⬝ᵥ (BilinForm.toMatrix b B) *ᵥ (b.repr y) := by
+  nth_rw 1 [← b.sum_repr x, ← b.sum_repr y]
+  simp [dotProduct, Matrix.mulVec_eq_sum, Finset.mul_sum]
```

Comment:
Just flagging that this is a [non-terminal simp](https://leanprover-community.github.io/extras/simp.html#non-terminal-simps) which makes for a brittle proof and indeed I encountered this in #27605

I presume this just slipped through since it's hard to spot but I just thought I'd highlight it in case @EtienneC30 hadn't seen this discussion before.